include/envoy/http/codec.h

```diff
@@ -132,6 +142,14 @@ class Stream {
    * @param reason supplies the reset reason.
    */
   virtual void resetStream(StreamResetReason reason) PURE;
+
+  /**
+   * Enable/disable further data from this stream.
+   * Cessation of data may not be immediate.  For example, for HTTP/2 this may stop further flow
+   * control window updates which will result in the peer eventually stopping sending data.
```

Comment:
It would avoid the double buffer and given the proximity between buffer and data source I think it'd be a nice clean thing to do, but it would make this one buffer + data source combination special and unusual so I'm not convinced it'd be worth doing unless I get to do the full flow control redesign we've talked about in person where everything seemlessly avoids the double buffering.